aerogear/mobile-security-service-operator
# fix(oauth): allow server access to the operator

## Motivation
[JIRA Comment](https://issues.jboss.org/browse/AEROGEAR-8987?focusedCommentId=13732533&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13732533)

Please note there are some changes here to the creation and deletion logic. These can be removed based on the other changes being made by @camilamacedo86 in another pr.

## What
An additional Service is added by the operator which is not fronted by a Route to be used by the operator in order to interact with the service as required by the consumption pr.

## Verification Steps

1. Using this image: `quay.io/laurafitzgerald/mobile-security-service-operator:0.1.0-dev-ag8987-2` 
2. Run `make create-all`
3. Verify access to the server and additionally that the make delete and update of apps is working correctly.
a) a new Service is created titled `security-service-server` which can be seen in the console
b) Operator updates the app in the service as required by the state of the app cr
   i) create app
   `oc new-project mobile-security-service-apps`
   `make create-app`
   Verify the app has been created in the UI
   ii) update app
   Update the name in the app cr
   Verify the app name is updated in the UI
   iii) delete app
  `make delete-app`
   Verify the app has been deleted from the UI

## Checklist:
